src/app.js

```diff
@@ -0,0 +1,123 @@
+go.app = function() {
+    var vumigo = require('vumigo_v02');
+    var _ = require('lodash');
+    var App = vumigo.App;
+    var EndState = vumigo.states.EndState;
+    var FreeText = vumigo.states.FreeText;
+    // TODO make menu state as start state with option to reset, resume, etc
+
+    var MomSpeak = App.extend(function(self) {
+        App.call(self, 'states_start');
+
+        self.init = function() {
+            if(!self.im.user.metadata.session_id)
+              self.im.user.metadata.session_id = vumigo.utils.uuid();
+        };
+
+        self.states.add('states_start', function(name, opts) {
+            return self.states.create('states_converse', {
+                msg: "Welcome to MomSpeak!"
+            });
+        });
+
+        self.states.add('states_converse', function(name, opts) {
+            if(_.isEmpty(self.im.config.wit)) {
+                return self.states.create('states_noconfig_error');
+            }
+            return new FreeText(name, {
+                question: opts.msg,
+                next: function(user_input) {
+                          return {
+                              name: 'states_post',
+                              creator_opts: {
+                                  input: user_input
+                              }
+                          };
+                      }
+            });
+        });
+
+        // takes user opts.input
+        self.states.add('states_post', function(name, opts) {
+            return go.utils.converse(self.im,self.im.user.metadata.session_id , opts.input)
+                    .then(function(wit_response) {
+                        return self.states.create('states_log', {
+                            data: wit_response.data,
+                            msg: opts.msg
+                        });
+            });
+        });
+
+        self.states.add('states_log', function(name, opts) {
+            self.im.log.debug(opts.data);
+            if("error" in opts.data) {
+                return self.states.create('states_wit_error');
+            }
+            if(opts.data === undefined) {
```

Comment:
The `undefined` check should come first, otherwise we'll hit the `"error" in opts.data` check which will throw a `TypeError`.